File: std/testdata/datastore/keygen/keygen.go
// Copyright 2022 Namespace Labs Inc; All rights reserved.
// Licensed under the EARLY ACCESS SOFTWARE LICENSE AGREEMENT
// available at http://github.com/namespacelabs/foundation

package main

import (
	"context"

	"namespacelabs.dev/foundation/framework/provisioning"
	"namespacelabs.dev/foundation/internal/planning/tool/protocol"
	"namespacelabs.dev/foundation/std/types"
	"namespacelabs.dev/go-ids"
)

func main() {
	provisioning.HandleInvoke(func(ctx context.Context, r provisioning.Request) (*protocol.InvokeResponse, error) {
		return &protocol.InvokeResponse{
			Resource: &types.Resource{
				Contents: []byte(ids.NewRandomBase32ID(128)),
			},
		}, nil
	})
}

File: std/testdata/datastore/denokeygen/keygen.ts
// Copyright 2022 Namespace Labs Inc; All rights reserved.
// Licensed under the EARLY ACCESS SOFTWARE LICENSE AGREEMENT
// available at http://github.com/namespacelabs/foundation

import { encode } from "https://deno.land/std@0.147.0/encoding/base64.ts";
import {
	readRequest,
	respond,
} from "https://namespacelabs.dev/foundation/std/experimental/deno/invocation.ts";

await readRequest();

respond({
	resource: {
		contents: encode(crypto.getRandomValues(new Uint8Array(128))),
	},
});
